lib/room.rb

```diff
@@ -0,0 +1,36 @@
+class Room
+  attr_reader :num_rooms, :array_of_used_rooms, :rate, :rooms
+
+  ROOMS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
```

Comment:
I like that you extracted the logic for manipulating the list of rooms to its own class. However, I'm not convinced by the name of this class - you've called it `Room` singular, but it tracks a collection of rooms. `Rooms` or `RoomList` might be a better option.